src/openrct2/world/MoneyEffect.cpp

```diff
@@ -79,66 +95,59 @@ void money_effect_create(money32 value)
         mapPosition.z = tile_element_height(mapPosition.x, mapPosition.y);
     }
     mapPosition.z += 10;
-    money_effect_create_at(-value, mapPosition.x, mapPosition.y, mapPosition.z, false);
+    CreateAt(-value, mapPosition.x, mapPosition.y, mapPosition.z, false);
 }
 
 /**
  *
  *  rct2: 0x00673232
  */
-void money_effect_update(rct_money_effect* moneyEffect)
+void rct_money_effect::Update()
 {
-    invalidate_sprite_2((rct_sprite*)moneyEffect);
-    moneyEffect->wiggle++;
-    if (moneyEffect->wiggle >= 22)
+    invalidate_sprite_2((rct_sprite*)this);
+    wiggle++;
+    if (wiggle >= 22)
     {
-        moneyEffect->wiggle = 0;
+        wiggle = 0;
     }
 
-    moneyEffect->move_delay++;
-    if (moneyEffect->move_delay < 2)
+    move_delay++;
+    if (move_delay < 2)
     {
         return;
     }
 
-    int32_t x = moneyEffect->x;
-    int32_t y = moneyEffect->y;
-    int32_t z = moneyEffect->z;
-    moneyEffect->move_delay = 0;
+    move_delay = 0;
 
-    if (moneyEffect->vertical)
+    if (vertical)
     {
         z += 1;
     }
     y += _moneyEffectMoveOffset[get_current_rotation()].y;
```

Comment:
This will be the source of your problem. This is adding the value onto the instance value instead of a new variable used just for sprite move.

Reply:
Thanks. That fixed the problem. I pushed the amended version (along with the suggested change for CreateAt).